server/app/controllers/applicant/ApplicantProgramBlocksController.java

```diff
@@ -513,18 +525,37 @@ private CompletionStage<Result> renderErrorOrRedirectToNextBlock(
               .createAddressQuestion();
 
       if (addressQuestion.needsAddressCorrection()) {
-        return applicantService
-            .getAddressSuggestionGroup(thisBlockUpdated)
-            .thenApplyAsync(
-                addressSuggestionGroup -> {
-                  ImmutableList<AddressSuggestion> suggestions =
-                      addressSuggestionGroup.getAddressSuggestions();
-                  String json = addressSuggestionJsonSerializer.serialize(suggestions);
-
-                  Boolean isEligibilityEnabledOnThisBlock =
-                      thisBlockUpdated.getLeafAddressNodeServiceAreaIds().isPresent();
-
-                  return ok(addressCorrectionBlockView.render(
+        AddressSuggestionGroup addressSuggestionGroup =
+            applicantService
+                .getAddressSuggestionGroup(thisBlockUpdated)
+                .toCompletableFuture()
+                .join();
+        ImmutableList<AddressSuggestion> suggestions =
+            addressSuggestionGroup.getAddressSuggestions();
+
+        AddressSuggestion[] suggestionMatch =
+            suggestions.stream()
+                .filter(suggestion -> suggestion.getAddress().equals(addressQuestion.getAddress()))
+                .toArray(AddressSuggestion[]::new);
+
+        if (suggestionMatch.length > 0) {
+          return confirmAddressWithSuggestions(
```

Comment:
The key here is that if there is a matching suggestion, we want to save it to the backend and then move on to the next screen without showing the address correction screen (calling `confirmAddressWithSuggestions` accomplishes this because it invokes `renderErrorOrRedirectToNextBlock` again, this time with a corrected address).